### PR TITLE
[move core modules] replace LibraCoin::T with Libra::T<LBR::T>

### PR DESCRIPTION
**Pull request by sblackshear**
- Introduce `Libra` module
- Introduce `LBR` module
- All uses of `LibraCoin` now use `Libra::T<LBR::T>`

The meat of this change is the `Libra` and `LBR` modules. Everything else is refactoring to support the changes there (in particular, to Move and MVIR functional tests).

## Motivation

Refactor coin mechanics to leverage Move generics.

### Have you read the [Contributing Guidelines on pull requests](https://github.com/libra/libra/blob/master/CONTRIBUTING.md#pull-requests)?

Y

## Test Plan

Updated all Move functional tests, which continue to pass.

### CODE REVIEW

File: language/ir-testsuite/tests/commands/return_type_mismatch_and_unused_resource.mvir [hunk 1]
```diff
@@ -1,12 +1,13 @@
 module M {
-    import 0x0.LibraCoin;
+import 0x0.LBR;
```
Comment (bob-wilson) [inline language/ir-testsuite/tests/commands/return_type_mismatch_and_unused_resource.mvir]: Bad indentation here.

File: language/ir-testsuite/tests/generics/coin_wrapper.mvir [hunk 2]
```diff
@@ -1,15 +1,16 @@
 module M {
-    import 0x0.LibraCoin;
+import 0x0.LBR;
```
Comment (bob-wilson) [inline language/ir-testsuite/tests/generics/coin_wrapper.mvir]: Another bad indentation.

File: language/ir-testsuite/tests/module_member_types/resource_has_resource_field.mvir [hunk 3]
```diff
@@ -1,4 +1,5 @@
 module Test {
-    import 0x0.LibraCoin;
-    resource T{fint: LibraCoin.T}
+import 0x0.LBR;
```
Comment (bob-wilson) [inline language/ir-testsuite/tests/module_member_types/resource_has_resource_field.mvir]: Another bad indentation

File: language/ir-testsuite/tests/module_member_types/resource_instantiate_bad_type.mvir [hunk 4]
```diff
@@ -1,9 +1,10 @@
 module Test {
-    import 0x0.LibraCoin;
+import 0x0.LBR;
```
Comment (bob-wilson) [inline language/ir-testsuite/tests/module_member_types/resource_instantiate_bad_type.mvir]: ...and another

File: language/ir-testsuite/tests/module_member_types/unrestricted_has_resource_field.mvir [hunk 5]
```diff
@@ -1,10 +1,11 @@
 // check: INVALID_RESOURCE_FIELD
 
 module Test {
-    import 0x0.LibraCoin;
+import 0x0.LBR;
```
Comment (bob-wilson) [inline language/ir-testsuite/tests/module_member_types/unrestricted_has_resource_field.mvir]: ...and another

File: language/ir-testsuite/tests/mutation/assign_resource_type.mvir [hunk 6]
```diff
@@ -1,18 +1,19 @@
 module A {
-    import 0x0.LibraCoin;
-    resource T {fr: LibraCoin.T}
+import 0x0.LBR;
```
Comment (bob-wilson) [inline language/ir-testsuite/tests/mutation/assign_resource_type.mvir]: ...and another

File: language/ir-testsuite/tests/mutation/destroy_resource_holder.mvir [hunk 7]
```diff
@@ -1,26 +1,28 @@
 module A {
-    import 0x0.LibraCoin;
-    resource A { c: LibraCoin.T }
-    public new(c: LibraCoin.T): Self.A {
+import 0x0.LBR;
```
Comment (bob-wilson) [inline language/ir-testsuite/tests/mutation/destroy_resource_holder.mvir]: ...and another

File: language/ir-testsuite/tests/mutation/mint_money.mvir [hunk 8]
```diff
@@ -1,23 +1,24 @@
 // check: "Unbound field balance"
 
 module Hack {
-    import 0x0.LibraCoin;
+import 0x0.LBR;
```
Comment (bob-wilson) [inline language/ir-testsuite/tests/mutation/mint_money.mvir]: ...and another

File: language/ir-testsuite/tests/mutation/mutate_resource_holder.mvir [hunk 9]
```diff
@@ -1,21 +1,22 @@
 module A {
-    import 0x0.LibraCoin;
-    resource A { c: LibraCoin.T }
-    public new(c: LibraCoin.T): Self.A {
+import 0x0.LBR;
```
Comment (bob-wilson) [inline language/ir-testsuite/tests/mutation/mutate_resource_holder.mvir]: ...and another

File: language/ir-testsuite/tests/mutation/unused_resource_holder.mvir [hunk 10]
```diff
@@ -1,31 +1,33 @@
 // check: RET_UNSAFE_TO_DESTROY_ERROR
 
 module A {
-    import 0x0.LibraCoin;
+import 0x0.LBR;
```
Comment (bob-wilson) [inline language/ir-testsuite/tests/mutation/unused_resource_holder.mvir]: ...and another

File: language/ir-testsuite/tests/natives/event_no_resource.mvir [hunk 11]
```diff
@@ -1,16 +1,17 @@
 module ResourceHandle {
     import 0x0.LibraAccount;
-    import 0x0.LibraCoin;
+import 0x0.LBR;
```
Comment (bob-wilson) [inline language/ir-testsuite/tests/natives/event_no_resource.mvir]: ...and another

File: language/ir-testsuite/tests/natives/vector/vector_module.mvir [hunk 12]
```diff
@@ -1,58 +1,60 @@
 module M {
-    import 0x0.LibraCoin;
+import 0x0.LBR;
```
Comment (bob-wilson) [inline language/ir-testsuite/tests/natives/vector/vector_module.mvir]: ...and another

File: language/ir-testsuite/tests/payment_channel/payment_channel.mvir [hunk 13]
```diff
@@ -1,6 +1,7 @@
 module PaymentChannel {
     import 0x0.Signature;
-    import 0x0.LibraCoin;
+import 0x0.LBR;
```
Comment (bob-wilson) [inline language/ir-testsuite/tests/payment_channel/payment_channel.mvir]: ...and another

File: language/ir-testsuite/tests/payment_channel/payment_channel_no_fixed_counterparty.mvir [hunk 14]
```diff
@@ -1,6 +1,7 @@
 module PaymentChannel {
     import 0x0.Signature;
-    import 0x0.LibraCoin;
+import 0x0.LBR;
```
Comment (bob-wilson) [inline language/ir-testsuite/tests/payment_channel/payment_channel_no_fixed_counterparty.mvir]: ...and another

File: language/ir-testsuite/tests/payment_channel/payment_channel_no_fixed_counterparty_multi_hop.mvir [hunk 15]
```diff
@@ -1,6 +1,7 @@
 module PaymentChannel {
     import 0x0.Signature;
-    import 0x0.LibraCoin;
+import 0x0.LBR;
```
Comment (bob-wilson) [inline language/ir-testsuite/tests/payment_channel/payment_channel_no_fixed_counterparty_multi_hop.mvir]: ...and another

File: language/ir-testsuite/tests/publish/publish_module_and_use.mvir [hunk 16]
```diff
@@ -1,40 +1,42 @@
 module MoneyHolder {
-        import 0x0.LibraCoin;
+import 0x0.LBR;
```
Comment (bob-wilson) [inline language/ir-testsuite/tests/publish/publish_module_and_use.mvir]: ...and another

File: language/ir-testsuite/tests/publish/publish_two_modules.mvir [hunk 17]
```diff
@@ -1,25 +1,26 @@
 module MoneyHolder {
-        import 0x0.LibraCoin;
+import 0x0.LBR;
```
Comment (bob-wilson) [inline language/ir-testsuite/tests/publish/publish_two_modules.mvir]: ...and another

File: language/ir-testsuite/tests/wallets/wallet_module.mvir [hunk 18]
```diff
@@ -70,13 +70,14 @@ module ColdWallet {
     import 0x0.AddressUtil;
     import 0x0.U64Util;
     import 0x0.Hash;
-    import 0x0.LibraCoin;
+import 0x0.LBR;
```
Comment (bob-wilson) [inline language/ir-testsuite/tests/wallets/wallet_module.mvir]: ...and another

File: language/move-prover/bytecode-to-boogie/tests/goldenfiles/libra_account.bpl [hunk 19]
```diff
@@ -1534,7 +1534,7 @@ ensures old(b#Boolean(Boolean(!(b#Boolean(ExistsResource(__m, LibraAccount_T_typ
     // declare local variables
     var __t2: Value; // AddressType()
     var __t3: Value; // LibraCoin_T_type_value()
-    var __t4: Value; // ByteArrayType()
+    var __t4: Value; // Vector_T_type_value(IntegerType())
```
Comment (bob-wilson) [inline language/move-prover/bytecode-to-boogie/tests/goldenfiles/libra_account.bpl]: I guess this is unrelated to this change and just fixing a leftover problem from converting bytearray to vector<u8>, right?

Comment (sblackshear) [reply language/move-prover/bytecode-to-boogie/tests/goldenfiles/libra_account.bpl]: Good catch--I guess somewhat forget to regenerate the goldenfiles and CI didn't catch it (since it only runs on changes in the prover directory). Seems ok to fix it here.
